src/components/fields.tsx

```diff
@@ -1,38 +1,41 @@
 import React from 'react';
 import { AsyncSelect, InlineField, Select } from '@grafana/ui';
 import { SelectableValue } from '@grafana/data';
-import { AutoCompleteEditorProps, EditorProps } from './types';
+import { AutoCompleteEditorProps, AutoCompleteConfig, EditorProps } from './types';
 import { get, update as _update, cloneDeep } from 'lodash';
 import { DataSource } from '../DataSource';
 import { combinedDesc } from 'graphspecs';
+import { MyQuery, ResponseDataAutocomplete } from 'types';
 
-const update = (x: any, path: string, func: any) => {
-  let copy = cloneDeep(x);
+const update = (x: MyQuery, path: string, func: () => SelectableValue<string> | string | undefined) => {
+  const copy = cloneDeep(x);
   _update(copy, path, func);
   return copy;
 };
 
-export const vsAutocomplete = (datasource: DataSource, autocompleteConfig: any) => (inputValue: string) =>
-  datasource
-    .restRequest('ajax_vs_autocomplete.py', {
-      ...autocompleteConfig,
-      value: inputValue.trim(),
-    })
-    .then((result) =>
-      result.data.result.choices.map(([value, label]: [string, string]) => ({
-        value,
-        label,
-        isDisabled: value === null,
-      }))
-    );
+export const vsAutocomplete =
+  (datasource: DataSource, autocompleteConfig: AutoCompleteConfig) =>
+  (inputValue: string): Promise<Array<{ value: string; label: string; isDisabled: boolean }>> =>
+    datasource
+      .restRequest<ResponseDataAutocomplete>('ajax_vs_autocomplete.py', {
```

Comment:
Someday, we need to think about the naming choice here. From what I can see, this is not much of a REST request at all, but an old-fashioned WebApi request.